src/components/ItemsList.js

```diff
@@ -104,12 +106,28 @@ const ItemsList = () => {
                   .toLowerCase()
                   .includes(searchTerm.toLowerCase()),
               )
+              .sort((a, b) =>
+                a.daysLeftForNextPurchase > b.daysLeftForNextPurchase
+                  ? 1
+                  : a.daysLeftForNextPurchase === b.daysLeftForNextPurchase
+                  ? a.shoppingListItemName > b.shoppingListItemName
+                    ? 1
+                    : -1
+                  : -1,
+              )
               .map((shoppingItemObject, index) => {
                 const shopIndex = shoppingList[0].items.indexOf(
                   shoppingItemObject,
                 );
                 return (
-                  <li key={shoppingItemObject.shoppingListItemName + index}>
+                  <li
+                    key={shoppingItemObject.shoppingListItemName + index}
+                    style={{
+                      backgroundColor: stylesFnx(
```

Comment:
This is great how you've used the function to return the background colour!

Reply:
I agree using a function makes the code cleaner and less congested. I would recommend  giving the function a more descriptive name rather than `stylesFnx`. Something like `getShoppingItemBackgroundStyles` will help one understand what the function does instantly.